exasol_integration_test_docker_environment/lib/docker/images/create/utils/file_directory_list_hasher.py

```diff
@@ -118,7 +118,16 @@ def collect_files_and_directories(
         tmp_collected_files = []
         if self.hash_directory_names:
             tmp_collected_directories.append((directory, directory))
+        inodes = set()
+        numIterations = 0
         for root, dirs, files in os.walk(directory, topdown=True, followlinks=self.followlinks):
+            stat = os.stat(root)
+            if stat.st_ino in inodes:
+                raise OSError(f"Directory: {directory} contains symlink loops (Symlinks pointing to a parent directory). Please fix!")
+            numIterations += 1
+            if numIterations > self.MAX_DIR_ITERATIONS:
```

Comment:
This doesn't work, because you can have multiple roots on the same depth. You need to check the number of the path segments of root instead. For example, by counting the `/` in root or by using len(pathlib.Path(root).parts)

Reply:
Regarding that check, at least in my unit test I get an error before reaching that limit:
```
OSError: [Errno 40] Too many levels of symbolic links: '/tmp/tmp9gyr6yf6/TestSymlinkLoops/level0/level1_4/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink/level2_4/level3_4/level4_4/evil_symlink'
```